packages/@aws-cdk/aws-amplify/lib/app.ts

```diff
@@ -0,0 +1,287 @@
+      description: props.description,
+      environmentVariables: this.environmentVariablesResolver,
+      name: props.name,
+      oauthToken: props.oauthToken,
+      repository: props.repository,
+      tags: props.tags
+    });
+
+    if (props.iamServiceRole) {
+      resource.iamServiceRole = props.iamServiceRole.roleArn;
+    }
+
+    if (props.buildSpec && !props.iamServiceRole) {
+      // create lazy role
+      const role = new LazyRole(scope, 'lazy-role', {
+        assumedBy: new ServicePrincipal('amplify.amazonaws.com')
+      });
+
+      resource.iamServiceRole = role.roleArn;
+    }
+
+    this.appArn = resource.attrArn,
+    this.appDefaultDomain = resource.attrDefaultDomain;
+    this.appId = resource.attrAppId;
+    this.appName = resource.attrAppName;
+
+    this.resource = resource;
+  }
+
+  public validate(): string[] {
+    const errors: string[] = [];
+
+    if (this.resource.buildSpec && !this.resource.iamServiceRole) {
+      errors.push(`You need to specify a service role when using buildspecs`);
+    }
+
+    return errors;
+  }
+
+  /**
+   * Add Service Role
+   *
+   * @param role
+   */
+  public addServiceRole(role: IRole) {
+    this.resource.iamServiceRole = role.roleArn;
+  }
+
+  /**
+   * Set Basic Auth on App
+   *
+   * @param username
+   * @param password
+   */
+  public setBasicAuth(username: string, password: string) {
+    this.basicAuthResolver.basicAuthConfig({
+      enableBasicAuth: true,
+      password,
+      username
+    });
+  }
+
+  /**
+   * Add Environment Variable to App
+   *
+   * @param name
+   * @param value
+   */
+  public addEnvironmentVariable(name: string, value: string) {
+    this.environmentVariablesResolver.addEnvironmentVariable(name, value);
+  }
+}
+
+/**
+ * App Interface
+ */
+export interface IApp extends IResource {
+  /**
+   * App ID
+   *
+   * @attribute
+   */
+  readonly appId: string;
+}
+
+/**
+ * App Options
+ */
+export interface AppOptions extends ResourceProps {
+  /**
+   * App Name
+   *
+   * @todo This might cause an error, it conflicts with the construct
+   *
+   * @default
+   */
+  readonly appName?: string;
+}
+
+/**
+ * App Properties
+ */
+export interface AppProps extends AppOptions {
+  /**
+   * Access Token
+   *
+   * @default
+   */
+  readonly accessToken?: string;
+
+  /**
+   * Basic Auth Config
+   *
+   * @default
+   */
+  readonly basicAuthConfig?: BasicAuthConfig;
+
+  /**
+   * BuildSpec
+   *
+   * @default
+   */
+  readonly buildSpec?: string;
+
+  /**
+   * Custom Rules
+   *
+   * @default
+   */
+  readonly customRules?: CustomRule[];
+
+  /**
+   * Description
+   *
+   * @default
+   */
+  readonly description?: string;
+
+  /**
+   * Environment Variables
+   *
+   * @default
+   */
+  readonly environmentVariables?: EnvironmentVariable[];
+
+  /**
+   * IAM Service Role
+   *
+   * @default
+   */
+  readonly iamServiceRole?: IRole;
+
+  /**
+   * Name
+   */
+  readonly name: string;
```

Comment:
If this is the physical name. It should be called “appName” and wired up to the base resource PhysicalName property

Reply:
It's not, you can create many App resources that have the same name in the same template